skpro/regression/linear/_glm.py

```diff
@@ -0,0 +1,387 @@
+
+    _tags = {
+        "authors": ["julian-fong"],
+        "maintainers": ["julian-fong"],
+        "python_version": None,
+        "python_dependencies": None,
+        "capability:multioutput": False,
+        "capability:missing": False,
+        "X_inner_mtype": "pd_DataFrame",
+        "y_inner_mtype": "pd_DataFrame",
+    }
+
+    def __init__(
+        self,
+        family=None,
+        offset=None,
+        exposure=None,
+        freq_weights=None,
+        var_weights=None,
+        missing="none",
+        start_params=None,
+        maxiter=100,
+        method="IRLS",
+        tol=1e-8,
+        scale=None,
+        cov_type="nonrobust",
+        cov_kwds=None,
+        use_t=None,
+        full_output=True,
+        disp=False,
+        max_start_irls=3,
+        add_constant=False,
+    ):
+        self.family = Gaussian()
+        self.offset = offset
+        self.exposure = exposure
+        self.freq_weights = freq_weights
+        self.var_weights = var_weights
+        self.missing = missing
+        self.start_params = start_params
+        self.maxiter = maxiter
+        self.method = method
+        self.tol = tol
+        self.scale = scale
+        self.cov_type = cov_type
+        self.cov_kwds = cov_kwds
+        self.use_t = use_t
+        self.full_output = full_output
+        self.disp = disp
+        self.max_start_irls = max_start_irls
+        self.add_constant = add_constant
+
+        super().__init__()
+
+    def _fit(self, X, y):
+        """Fit regressor to training data.
+
+        Writes to self:
+            Sets fitted model attributes ending in "_".
+
+        Parameters
+        ----------
+        X : pandas DataFrame
+            A (n,k) array where n is the number of observations and k is the number
+            of regressors. An intercept is not included by default and should be
+            added by the user (models specified using a formula include an
+            intercept by default). Equivalent to statsmodel's (exog).
+
+        y : pandas DataFrame
+            1d array of the endogenous response variable. This array can be 1d or
+            2d. Binomial family models accept a 2d array  with two columns.
+            If supplied each observation is expected to  be [success, failure].
+            Equivalent to statsmodel's (endog).
+
+        Returns
+        -------
+        self : reference to self
+        """
+        if self.add_constant:
+            X = add_constant(X)
+
+        y_col = y.columns
+
+        glm_estimator = GLM(
+            endog=y,
+            exog=X,
+            family=self.family,
+            offset=self.offset,
+            exposure=self.exposure,
+            freq_weights=self.freq_weights,
+            var_weights=self.var_weights,
+            missing=self.missing,
+        )
+
+        self._estimator = (
+            glm_estimator  # does this need to be cloned using some clone method?
+        )
+
+        fitted_glm_model = self._estimator.fit(
```

Comment:
I would write this to `glm_fitted_`, and describe in the `Attributes` section of the docstring.

Reply:
Should I include forward both the glm estimator ```_estimator``` and the glm fitted estimator (now named ```glm_fit_```) and their corresponding attributes? I will include everything inside the docstring as well.

Reply:
yes, see below